python/amici/sbml_import.py

```diff
@@ -435,8 +438,16 @@ def _gather_locals(self) -> None:
                 self.local_symbols[r.getId()] = _get_identifier_symbol(r)
 
         # SBML time symbol + constants
-        self.local_symbols['time'] = symbol_with_assumptions('time')
-        self.local_symbols['avogadro'] = symbol_with_assumptions('avogadro')
+        for symbol_name in ['time', 'avogadro']:
+            if symbol_name in self.local_symbols:
+                # Supporting this is probably kinda tricky and this sounds
+                # like a stupid thing to do in the first place.
+                raise SBMLException(f'AMICI does not support SBML models '
+                                    f'containing variables with Id '
+                                    f'{symbol_name}.')
```

Comment:
Certainly happens though.. this caused an issue for a student because a model from BioModels included a transformation of time as a parameter assignment rule with Id `Time` :see_no_evil: 

This exception is good and would have made debugging much easier. Is it already -- or, should it be -- case-insensitive?

Reply:
I believe everything in sympy is case sensitive and as the source of this problem is name collision during sympification, the check shouldn't be case sensitive.

Fixing this is really nasty though as we couldn't use `formulaToL3String` anymore and would have to start parsing xml in order to be able to differentiate between `csymbols` and regular symbol ids. I am pretty sure that in terms of SBML support, there are things with much better time investment vs coverage/relevance ratio.